### PR TITLE
Add sniper

### PR DESCRIPTION
**Pull request by olliy1or**

### CODE REVIEW

**Review by Panquesito7 — APPROVED** [review]

File: mods/ctf/ctf_combat/ctf_ranged/init.lua [hunk 1]
```diff
@@ -233,3 +275,71 @@ ctf_ranged.simple_register_gun("ctf_ranged:smg", {
...
+	time = time + dtime
+	if time < timer then
+		return
+	end
+
+	time = 0
+	for name, original_item in pairs(scoped) do
+		local player = minetest.get_player_by_name(name)
+		if not player then
+			scoped[name] = nil
```
Comment (savilli) [inline mods/ctf/ctf_combat/ctf_ranged/init.lua:337]: You can't modify the map while iterating

**Review by savilli — CHANGES_REQUESTED** [review]

File: mods/ctf/ctf_combat/ctf_ranged/init.lua [hunk 2]
```diff
@@ -1,7 +1,11 @@
 ctf_ranged = {}
-
+local hud = mhud.init()
 local shoot_cooldown = ctf_core.init_cooldowns()
 
+local scoped = {}
+local scale_const = 6
+local timer = 0.2
```
Comment (savilli) [inline mods/ctf/ctf_combat/ctf_ranged/init.lua:7]: I think 0.2 is too low. 1 second is fine.

File: mods/ctf/ctf_combat/ctf_ranged/init.lua [hunk 3]
```diff
@@ -233,3 +278,71 @@ ctf_ranged.simple_register_gun("ctf_ranged:smg", {
...
+	time = time + dtime
+	if time < timer then
+		return
+	end
+
+	time = 0
+	for name, original_item in pairs(scoped) do
+		local player = minetest.get_player_by_name(name)
+		if not player then
+			scoped[name] = nil
```
Comment (savilli) [inline mods/ctf/ctf_combat/ctf_ranged/init.lua:340]: No

File: mods/ctf/ctf_combat/ctf_ranged/init.lua [hunk 4]
```diff
@@ -178,6 +182,47 @@ function ctf_ranged.simple_register_gun(name, def)
 	end))
 end
 
+minetest.register_on_leaveplayer(function(player)
+	if scoped[player:get_player_name()] then
```
Comment (savilli) [inline mods/ctf/ctf_combat/ctf_ranged/init.lua:186]: You can do just
```lua
scoped[player:get_player_name()] = nil
```

**Comment by savilli** [comment]
LGTM

**Pull request merged**